lib/csv.rb

```diff
@@ -1363,7 +1379,8 @@ def writer
   end
 
   def writer_options
-    @writer_options.merge(header_fields_converter: header_fields_converter)
+    @writer_options.merge(header_fields_converter: header_fields_converter,
+                          writer_fields_converters: writer_fields_converter)
```

Comment:
Could you use `fields_converter`? `writer_` is redundant for writer. It's not plural.